app/models/course.rb

```diff
@@ -226,4 +226,26 @@ def publish_enrichment(current_user)
       enrichment.publish(current_user)
     end
   end
+
+  def toggle_site(site_to_toggle)
+    is_course_new = new? # persist this before we change anything
+
+    site_status = site_statuses.find_or_initialize_by(site: site_to_toggle) do |ss|
+      ss.vac_status = SiteStatus.default_vac_status_given(study_mode: study_mode)
```

Comment:
I just realised there may be a better way to do this: move this block into an `after_initialize` filter on `SiteStatus`.

Reply:
Fixed in bb0cbf9